src/main/java/sergey/EquationServlet.java

```diff
@@ -0,0 +1,194 @@
+package sergey;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.*;
+import java.util.zip.DataFormatException;
+
+import static javax.servlet.http.HttpServletResponse.*;
+
+@WebServlet(
+        urlPatterns = {"/calc/*"}
+)
+public class EquationServlet extends HttpServlet {
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        HttpSession session = req.getSession();
+        String key = req.getPathInfo().replaceAll("/", "");
+        String val = req.getReader().readLine();
+        try {
+            if (key.equals("equation")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4723796)